docs/docs/guides/customization/behavior.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Behavior Customization
+---
+
+In this guide, you will create a [ZITADEL action](../../concepts/features/actions).
+After users register using an external identity provider, the action assigns them a role.
+
+## Prerequisites
+
+Before you start, make sure you have everything set up correctly.
+
+- You need to be at least a ZITADEL *ORG_OWNER*
+- Your ZITADEL organization needs to have the actions feature enabled. <!-- TODO: How to enable it for SaaS ZITADEL? -->
+- [Your ZITADEL organization needs to have at least one external identity provider enabled](../authentication/identity-brokering)
+- [You need to have at least one role configured for a project](../basics/projects)
+
+## Copy some information for the action
+
+1. Select the **Projects** navigation item.
+1. Select a project that has a role configured.
+1. Copy the projects **Resource Id** on the screens top right.
+1. Scroll to the **ROLES** section and note some roles key.
+
+## Create the action
+
+1. Select the **Actions** navigation item.
+1. In the **Actions <i class="las la-code"></i>** section, select the **+ New** button.
+1. Give the new action the name `addGrant`.
+1. Paste this snippet into the multiline textfield.
+1. Replace the snippets placeholders and select **Save**.
+
+```js
+function addGrant(ctx, api) {
+  api.userGrants.push({
+    ProjectID: '<the projects resource ID you copied above>',
+    Roles: ['<the role key you copied above>']
+  });
+}
+```
+
+## Run the action when a user registers
+
+Now, make the action hook into the [external authentication flow](../../apis/actions#external-authentication-flow).
+
+1. In the **Flows <i class="las la-exchange-alt"></i>** section, select the **+ New** button.
+1. Select the **Flow Type** *External Authentication*.
+1. Select the **Trigger Type** *Post Creation*.
+1. In the **Actions** dropdown, check *addGrant*.
+1. Select the **Save** button.
+
+<!-- TODO: ## Test if your action works -->
```

Comment:
I think we should wait for v2 to complete this

Reply:
fine by me